README.md

```diff
@@ -0,0 +1,214 @@
+
+## Installation
+
+**Requires go 1.14**
+
+Install the module in your package or app with `go get "github.com/filecoin-project/go-ds-versioning/datatransfer"`
+
+## Usage
+
+For our example, let's imagine a key value store of records with the following structures:
+
+```golang
+type FruitType string
+
+type FruitBasketOld struct {
+    Type FruitType
+    Count uint64
+    Price big.Int
+}
+```
+
+We want to support multiple fruits in the basked so our new structure is as follows:
+
+```golang
+type FruitBasket struct {
+    Types []FruitType
+    Count uint64
+    Price big.Int
+}
+```
+
+We've generated cbor-gen code for both of these. Now we have a function to migrate one to the other:
+
+```golang
+func MigrateFruitBasket(old * FruitBasketOld) (*FruitBasket, error) {
+    return &FruitBasket{
+        Types: []FruitType{old.Type},
+        Count: old.Count,
+        Price: old.Price,
+    }, nil
+}
+```
+
+### Defining Migrations
+
+Now we want to migrate our store. Let's say we've never migrated our store before, so this is the first time we're going to migrate it.
+
+Let's defined a initial set of migration using go-ds-versioning's builders:
+
+```golang
+import (
+	versioning "github.com/filecoin-project/go-ds-versioning/pkg"
+	"github.com/filecoin-project/go-ds-versioning/pkg/versioned"
+)
+
+builder := versioned.NewVersionedBuilder(MigrateFruitBasket, versioning.VersionKey("1"))
+
+migrationBuilders := versioned.BuilderList{builder}
+
+migrations, err := migrationBuilders.Build()
+```
+
+There's a lot happening there so let's take a look at this. We build a migration here:
+
+```golang
+builder := versioned.NewVersionedBuilder(MigrateFruitBasket, versioning.VersionKey("1")),
+```
+
+We're defining a migration that will use MigrateFruitBasket as its transformation function and will migration to version `1`. While not shown here, there are other things we can also do with this builder:
+
+```golang
+// if we want to provide a mechanism to reverse this migration
+builder := builder.Reversable(UnMigrateFruitBasket)
+
+// after our initial version, we always want to make sure we define the old version we're migrating from
+version2Builder := version2Builder.OldVersion("1")
+
+// we may need to for whatever reason, leave certain keys out of a migration
+// (i.e. delete them, or we may need to exclude some keys that are at the same
+// namespace hierarchy in our initial migration)
+builder := builder.FilterKeys("rotten-fruit-basket")
+```
+
+We're assuming we'll probably define all our migrations in one place, so we make a `BuilderList` -- a list of migration definitions assembled using our builder interface. Typically you can just put your builders inline in the BuilderList.
+
+Finally, to turn these into actual migrations, we call `.Build()` on the `BuilderList` -- this will that our migrations are valid. In particular, every migration function must have the form:
```

Comment:
```suggestion
Finally, to turn these into actual migrations, we call `.Build()` on the `BuilderList` -- this will ensure that our migrations are valid. In particular, every migration function must have the form:
```